contracts/contracts/mcms/rbac_timelock.tolk

```diff
@@ -686,7 +723,9 @@ fun Timelock<T>.executeBatch(
     predecessor: uint256,
     salt: uint256
 ): void {
-    self.onlyRoleOrAdminRole(EXECUTOR_ROLE, sender);
+    if (self.data.executorRoleCheckEnabled) {
```

Comment:
If anybody has strong preferences, this can be extracted in a separate contract using a hook